src/function.rs

```diff
@@ -903,6 +907,24 @@ impl Function {
     (ret >= 0).then_some(ret as u32)
   }
 
+  #[inline(always)]
+  pub fn get_script_origin(&self) -> &ScriptOrigin {
+    unsafe {
+      let ptr = v8__Function__GetScriptOrigin(self);
+      &*ptr
+    }
+  }
+
+  /// Returns scriptId.
+  #[inline(always)]
+  pub fn get_script_id(&self) -> Option<u32> {
```

Comment:
Should this be i32?

Reply:
Was unsure about this. V8 docs are sparse on when a script ID is < 0, but the only cases I could find in the source were 1) uninitialized ScriptOrigin classes (value is -1) and 2) kTemporaryScriptId = -2. Due to uncertainty on when these cases actually occur, I just treated these as None values.

I can change this to i32 instead if you think that would make more sense.

Reply:
@mmastrac Would you suggest i32 instead?

Reply:
Yes, we should use `i32` here. Also please remove `get_` prefix - in `v8/include/v8-function.h` this function is named `ScriptId()`

Reply:
Done @bartlomieju @mmastrac